services/controller/src/plz/controller/instances/aws/ec2_instance_group.py

```diff
@@ -95,13 +97,16 @@ def instance_initialization_code(self) -> str:
         ])
         return self._instance_initialization_code
 
-    def acquire_instance(
+    def run_in_instance(
```

Comment:
Bit weird to have a "release" without an "acquire". Is `release_instance` still used? If it's part of cleanup/harvesting, then let's rename it accordingly.